libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt

```diff
@@ -599,6 +615,146 @@ class MapboxNavigationTest {
         }
     }
 
+    @Test
+    fun `new routes are set after reroute`() {
+        val newRoutes = listOf(mockk<NavigationRoute>(), mockk())
+        val navigationRerouteController: NavigationRerouteController? = mockk(relaxed = true) {
+            every { reroute(any<NavigationRerouteController.RoutesCallback>()) } answers {
+                (firstArg() as NavigationRerouteController.RoutesCallback)
+                    .onNewRoutes(newRoutes, mockk())
+            }
+        }
+        coEvery {
+            tripSession.setRoutes(any(), any(), any())
+        } returns NativeSetRouteValue(emptyList())
+
+        createMapboxNavigation()
+        mapboxNavigation.setRerouteController(navigationRerouteController)
+        val observers = mutableListOf<OffRouteObserver>()
+        verify { tripSession.registerOffRouteObserver(capture(observers)) }
+
+        observers.forEach {
+            it.onOffRouteStateChanged(true)
+        }
+        coVerify(exactly = 1) {
+            directionsSession.setRoutes(newRoutes, 0, RoutesExtra.ROUTES_UPDATE_REASON_REROUTE)
```

Comment:
I'm wondering, do we need to check other side effects of set routes like `routeAlternativesController.processAlternativesMetadata`? 
@dzinad , @RingerJK , what do you think?